### DISCUSSION TITLE
WinRing0x64.sys file

### DISCUSSION THREAD

**Discussion by dominiktr** [seerge/g-helper#678] [General]
The WinRing0x64.sys file has a known vulnerability. 
https://www.cvedetails.com/cve/CVE-2020-14979/
Is it possible to replace it with a newer version? 
If I don't use undervolting, can I extract only GHelper.exe without the WinRing0x64.sys, GHelper.dll.config, and WinRing0x64.dll files?

**Comment by seerge** [comment]
Currently undervolting library (including WinRing0x64.sys) is literally taken from UXTU (as mentioned in readme), as my assumption was that it's fine as it's being used by lots of people w/o any problems.

I have tried to replace WinRing0x64.sys with a freshly assembled version from latest sources, and it doesn't seem to work at all then. 
Popular app Libre Hardware Monitor had similar case https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/issues/984

It seems that you need a EV certificate to sign that thing as well. So for now it will stay like that probably. (Unless you know a better solution?) :) 

To answer your question - if you don't want to use undervolting / CPU temp limit (i.e. "advanced" tab in fans+power) you don't need a dll/sys files.
